### PR TITLE
Migrate Max Dome

### PR DESCRIPTION
**Pull request by naheedsa**

### CODE REVIEW

File: indi-maxdomeii/maxdomeii.cpp [hunk 1]
```diff
@@ -210,15 +209,16 @@ void MaxDomeII::TimerHit()
 
     // Watch dog
     nTimeSinceLastCommunication++;
-    if (WatchDogNP.np[0].value > 0 && WatchDogNP.np[0].value <= nTimeSinceLastCommunication)
```
Comment (naheedsa) [inline indi-maxdomeii/maxdomeii.cpp:213]: @knro , there are few lines which are using WatchDogNP.np[0].value.
Is it count? i.e I commented these lines. Or we simply using normal values/
Please let me know so I will update in this PR

Comment (knro) [reply indi-maxdomeii/maxdomeii.cpp:213]: No, it's not count. It's the first element of this property. i.e. WatchDogNP[0].getValue()

Comment (naheedsa) [reply indi-maxdomeii/maxdomeii.cpp:213]: updated. Pushed a commit

File: indi-maxdomeii/maxdomeii.cpp [hunk 2]
```diff
@@ -581,14 +581,14 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
...
-    if (!strcmp(name, TicksPerTurnNP.name))
+    if (TicksPerTurnNP.isNameMatch(name))
     {
         double nVal;
         char cLog[255];
         int error;
         int nRetry = 3;
 
-        if (IUUpdateNumber(&TicksPerTurnNP, values, names, n) < 0)
+        if (TicksPerTurnNP.update(values, names, n) < 0)
```
Comment (knro) [inline indi-maxdomeii/maxdomeii.cpp:591]: .update is a boolean and does not return int

File: indi-maxdomeii/maxdomeii.cpp [hunk 3]
```diff
@@ -643,54 +645,58 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
...
     // Watch dog
     // ===================================
-    if (!strcmp(name, WatchDogNP.name))
+    if (WatchDogNP.isNameMatch(name))
     {
         double nVal;
         char cLog[255];
 
-        if (IUUpdateNumber(&WatchDogNP, values, names, n) < 0)
+        if (WatchDogNP.update(values, names, n) < 0)
```
Comment (knro) [inline indi-maxdomeii/maxdomeii.cpp:670]: same

File: indi-maxdomeii/maxdomeii.cpp [hunk 4]
```diff
@@ -701,21 +707,24 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
...
+                ShutterOperationAzimuthNP.apply();
             }
 
             return true;
         }
         // Incorrect value.
-        ShutterOperationAzimuthNP.s = IPS_ALERT;
-        IDSetNumber(&ShutterOperationAzimuthNP, "Invalid shutter operation azimuth position");
+        ShutterOperationAzimuthNP.setState(IPS_ALERT);
+        LOG_INFO("Invalid shutter operation azimuth position");
```
Comment (knro) [inline indi-maxdomeii/maxdomeii.cpp:726]: should be LOG_ERROR

Comment (naheedsa) [reply indi-maxdomeii/maxdomeii.cpp]: Updated

File: indi-maxdomeii/maxdomeii.cpp [hunk 5]
```diff
@@ -643,54 +647,59 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
...
 
         nVal = values[0];
         if (nVal >= 0 && nVal <= 3600)
         {
             sprintf(cLog, "New watch dog set: %lf", nVal);
-            WatchDogNP.s           = IPS_OK;
-            WatchDogNP.np[0].value = nVal;
-            IDSetNumber(&WatchDogNP, "%s", cLog);
+            WatchDogNP.setState(IPS_OK);
+            // FixMe
```
Comment (fcasarramona) [inline indi-maxdomeii/maxdomeii.cpp]: Also, pending.

File: indi-maxdomeii/maxdomeii.cpp [hunk 6]
```diff
@@ -701,21 +710,25 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
             if (error == IPS_OK)
             {
                 nShutterOperationPosition             = nVal;
-                ShutterOperationAzimuthNP.s           = IPS_OK;
-                ShutterOperationAzimuthNP.np[0].value = nVal;
-                IDSetNumber(&ShutterOperationAzimuthNP, "New shutter operation azimuth set");
+                ShutterOperationAzimuthNP.setState(IPS_OK);
+                // FixMe
```
Comment (fcasarramona) [inline indi-maxdomeii/maxdomeii.cpp]: Another pending Fix.

File: indi-maxdomeii/maxdomeii.cpp [hunk 7]
```diff
@@ -604,37 +605,40 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
                 sprintf(cLog, "New Ticks Per Turn set: %lf", nVal);
                 nTicksPerTurn    = nVal;
                 nHomeTicks       = floor(0.5 + nHomeAzimuth * nTicksPerTurn / 360.0); // Calculate Home ticks again
-                TicksPerTurnNP.s = IPS_OK;
-                TicksPerTurnNP.np[0].value = nVal;
-                IDSetNumber(&TicksPerTurnNP, "%s", cLog);
+                TicksPerTurnNP.setState(IPS_OK);
+                // FixMe
+                // TicksPerTurnNP.np[0].value = nVal;
```
[the remark this answers is not in the document]
Comment (naheedsa) [reply indi-maxdomeii/maxdomeii.cpp]: No such line exists in the latest code. This comment is outdated. and it is fixed.

File: indi-maxdomeii/maxdomeii.cpp [hunk 8]
```diff
@@ -643,54 +647,59 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
             sprintf(cLog, "New home azimuth set: %lf", nVal);
             nHomeAzimuth              = nVal;
             nHomeTicks                = floor(0.5 + nHomeAzimuth * nTicksPerTurn / 360.0);
-            HomeAzimuthNP.s           = IPS_OK;
-            HomeAzimuthNP.np[0].value = nVal;
-            IDSetNumber(&HomeAzimuthNP, "%s", cLog);
+            HomeAzimuthNP.setState(IPS_OK);
+            // HomeAzimuthNP.[0].value = nVal;
```
[the remark this answers is not in the document]
Comment (naheedsa) [reply indi-maxdomeii/maxdomeii.cpp:649]: Sorry for the issue. It is now updated.

File: indi-maxdomeii/maxdomeii.cpp (hunk 6 above)
Comment (naheedsa) [reply indi-maxdomeii/maxdomeii.cpp]: Same.
No such line of code exists. It is updated in the latest code.

File: indi-maxdomeii/maxdomeii.cpp (hunk 5 above)
Comment (naheedsa) [reply indi-maxdomeii/maxdomeii.cpp]: Same, this is an outdated file. It is already updated. i.e using WatchDogNP[0].setValue(nVal);

**Pull request merged**